components/domains/steps/userInfoForm.tsx

```diff
@@ -0,0 +1,174 @@
+import React, {
+  FunctionComponent,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import styles from "../../../styles/components/registerV3.module.css";
+import { FormContext } from "@/context/FormProvider";
+import { FormType } from "@/utils/constants";
+import { getDomainWithStark, isValidEmail } from "@/utils/stringService";
+import TextField from "../../UI/textField";
+import SwissForm from "../swissForm";
+import SelectIdentity from "../selectIdentity";
+import NumberTextField from "@/components/UI/numberTextField";
+import { useAccount } from "@starknet-react/core";
+import ConnectButton from "@/components/UI/connectButton";
+import Button from "@/components/UI/button";
+import RenewalDomainsBox from "../renewalDomainsBox";
+import { areDomainSelected } from "@/utils/priceService";
+
+type UserInfoFormProps = {
+  type: FormType;
+  goToNextStep: () => void;
+  imageUrl: string;
+};
+
+const UserInfoForm: FunctionComponent<UserInfoFormProps> = ({
+  type,
+  goToNextStep,
+  imageUrl,
+}) => {
+  const maxYearsToRegister = 25;
+  const { address } = useAccount();
+  const { formState, updateFormState } = useContext(FormContext);
+  const [email, setEmail] = useState<string>("");
+  const [emailError, setEmailError] = useState<boolean>(true);
+  const [selectedDomains, setSelectedDomains] =
+    useState<Record<string, boolean>>();
+
+  useEffect(() => {
+    if (type === FormType.REGISTER) return;
+    updateFormState({ selectedDomains });
+  }, [selectedDomains]);
+
+  const getTitle = () => {
+    switch (type) {
+      case FormType.REGISTER:
+        return getDomainWithStark(Object.keys(formState.selectedDomains)[0]);
```

Comment:
Could you add a comment explaining what you're doing here. 

Don't remember what Object.keys does and I'm not sure to understand the link with getting a title

Reply:
`formState.selectedDomains` is of type `Record<string, boolean>`. In the case of a registration of domain we only have one domain in our record so Object.keys allows to parse and get the first entry. In the design of the `userInfoForm` the title is the domain name the user is currently buying.